test/e2e/artifacts_test.go

```diff
@@ -327,6 +334,15 @@ func (s *ArtifactsSuite) TestArtifactGC() {
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-step-wf-tmpl-no-gc.yaml",
 			hasGC:                        false,
+			workflowShouldSucceed:        true,
+			expectedGCPodsOnWFCompletion: 0,
+			expectedArtifacts:            []artifactState{},
+		},
+		// Workflow fails to write an artifact that's been defined as an Output
+		{
+			workflowFile:                 "@testdata/artifactgc/artgc-artifact-not-written.yaml",
+			hasGC:                        true,
+			workflowShouldSucceed:        false, // artifact not being present causes Workflow to fail
 			expectedGCPodsOnWFCompletion: 0,
 			expectedArtifacts:            []artifactState{},
```

Comment:
Okay, I've modified the test Workflow so that it also successfully saves an artifact file and garbage collects it. Had to update the test to look for artifact keys names that are auto-generated.